src/prefect/artifacts.py

```diff
@@ -124,7 +124,28 @@ async def create_table_artifact(
     Returns:
         - The table artifact ID.
     """
-    formatted_table = json.dumps(table) if isinstance(table, list) else table
+    if isinstance(table, dict):
+        for _, v in table.items():
+            if isinstance(v, list):
+                if any(isinstance(x, float) and x != x for x in v):
+                    raise ValueError(
+                        "`create_table_artifact` does not support NaN values in `table`"
+                        " argument."
+                    )
+        formatted_table = table
+    elif isinstance(table, list):
+        try:
+            formatted_table = json.dumps(table, allow_nan=False)
+        except ValueError:
```

Comment:
Can we just cast NaN to a reasonable value instead of throwing an exception?

Reply:
The user's not giving us JSON though, they're giving us a table for display purposes in the UI. That it's invalid JSON is a detail of our implementation. We could easily make it valid JSON by replacing it with "NaN" and then it'd be displayed well in the UI.

Reply:
I agree

Reply:
@billpalombi do you have any thoughts on whether NaNs in table artifacts should be converted so that they can be rendered properly in the UI or whether we should raise an exception?

Reply:
I agree with you and @madkinsz, we should convert NaN's to something we can render rather than throwing an exception.